### PR TITLE
make units an enum

### PR DESCRIPTION
**Pull request by nreinicke**
Updates the units in hive a bit to support more ways to represent units as strings. Now, units are converted from a string into an Unit enum. This was done to support some internal work to integrate routee powertrain.

### CODE REVIEW

File: nrel/hive/util/units.py [hunk 1]
```diff
@@ -1,7 +1,40 @@
...
 
+from typing import Dict
+from enum import Enum, auto
+
+
+class Unit(Enum):
+    MPH = auto()
+    KMPH = auto()
+    MILES = auto()
+    KILOMTERS = auto()
```
Comment (robfitzgerald) [inline nrel/hive/util/units.py:11]: typo (missing 'E')

File: nrel/hive/model/vehicle/mechatronics/powertrain/tabular_powertrain.py [hunk 2]
```diff
@@ -43,16 +42,9 @@ def from_data(
...
-            raise TypeError(f"{data['speed_units']} not a recognized unit in hive")
-        elif not valid_unit(data["distance_units"]):
-            raise TypeError(f"{data['distance_units']} not a recognized unit in hive")
-        elif not valid_unit(data["energy_units"]):
-            raise TypeError(f"{data['energy_units']} not a recognized unit in hive")
-
-        speed_units = data["speed_units"]
-        energy_units = data["energy_units"]
-        distance_units = data["distance_units"]
+        speed_units = Unit.from_string(data["speed_units"])
```
Comment (robfitzgerald) [inline nrel/hive/model/vehicle/mechatronics/powertrain/tabular_powertrain.py:45]: try/catch and chain so we can tell user "failure reading tabular powertrain input" on Unit.from_string exception.

**Review by robfitzgerald — CHANGES_REQUESTED** [review]
dig it, certainly helps a bit without adding any overhead. a typo fix request, and a request for a try/catch block lies in your future. :bee:

File: nrel/hive/util/units.py (hunk 1 above)
Comment (nreinicke) [reply nrel/hive/util/units.py]: whoops! good catch!

File: nrel/hive/model/vehicle/mechatronics/powertrain/tabular_powertrain.py (hunk 2 above)
Comment (nreinicke) [reply nrel/hive/model/vehicle/mechatronics/powertrain/tabular_powertrain.py]: yes, that's a great idea, added in

**Review by robfitzgerald — DISMISSED** [review]
good stuff, thanks nick!

**Comment by nreinicke** [comment]
@robfitzgerald - sorry! I pushed up a change right as you were approving the PR. Do you mind taking a super brief look at [a2281be](https://github.com/NREL/hive/pull/120/commits/a2281be12bda5d49f247a2e5dcf50b6dbe569a79).

**Review by robfitzgerald — APPROVED** [review]
def valid_commit

**Pull request merged**